### PR TITLE
[Kotlin Cleanup] NoteEditor nullability cleanup

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
> com.ichi2.anki.FieldEditLineTest > testSaveRestore[test(AVD) - 10] FAILED 
	kotlin.UninitializedPropertyAccessException: lateinit property mEditorNote has not been initialized
	at com.ichi2.anki.NoteEditor.onCollectionLoaded(NoteEditor.kt:456)

seems to be a real error. Please correct.

**Comment by divyansh-dxn** [comment]
Sorry, forgot to mark as draft earlier, was just testing right now.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -136,17 +137,17 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
...
     private var mSelectedTags: ArrayList<String>? = null
 
     @get:VisibleForTesting
     var deckId: Long = 0
         private set
     private var mAllModelIds: ArrayList<Long>? = null
     @KotlinCleanup("this ideally should be Int, Int?")
-    private var mModelChangeFieldMap: MutableMap<Int, Int>? = null
-    private var mModelChangeCardMap: HashMap<Int, Int?>? = null
+    private lateinit var mModelChangeFieldMap: MutableMap<Int, Int>
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:149]: I'm sorry, I fear I have trouble reviewing this PR. 
Lots of things are occurring at the same time. This makes it hard to check for example that this is indeed a lateinit, that each use of this value occur after initialization, while a lot of things at the same time.
I'm fine with squashing at the end if each change are too small to be worth it. But I'd really prefer if you could make more atomic commits and move variables to lateinit one-by-one with potentially a one-line commit message explaining why it's valid.

Comment (divyansh-dxn) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:149]: Yes, I feared about that. I'll drop some commits and will make changes small.

Comment (divyansh-dxn) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:149]: Will do somewhat later, got some more things to do currently. Converting to draft.

**Comment by divyansh-dxn** [comment]
Resolved the conflicts, ready for review. Had been open for so long, wasn't much conflicting so went well.

**Review by david-allison — APPROVED** [review]

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**